tests/Metadata/Processor/IndexProcessorTest.php

```diff
@@ -0,0 +1,173 @@
+use Fazland\ODM\Elastica\Tests\Traits\DocumentManagerTestTrait;
+use PHPUnit\Framework\TestCase;
+
+class IndexProcessorTest extends TestCase
+{
+    use DocumentManagerTestTrait;
+
+    /**
+     * @var IndexProcessor
+     */
+    private $processor;
+
+    /**
+     * @var DocumentMetadata
+     */
+    private $documentMetadata;
+
+    protected function setUp()
+    {
+        $this->processor = new IndexProcessor();
+        $this->documentMetadata = new DocumentMetadata(new \ReflectionClass(Foo::class));
+    }
+
+    public function testAnalyzersAreProcessedCorrectly()
+    {
+        $index = new Index();
+
+        $analyzer = new Analyzer();
+        $analyzer->name = 'foo_name';
+        $analyzer->tokenizer = 'foo_tokenizer';
+
+        $index->analyzers = [$analyzer];
+
+        $this->processor->process($this->documentMetadata, $index);
+
+        $this->assertEquals([
+            'settings' => [
+                'analysis' => [
+                    'analyzer' => [
+                        'foo_name' => [
+                            'tokenizer' => 'foo_tokenizer',
+                        ],
+                    ],
+                ],
+            ],
+        ], $this->documentMetadata->indexParams);
+    }
+
+    public function testFiltersAreProcessedCorrectly()
+    {
+        $index = new Index();
+
+        $filter = new Filter();
+        $filter->name = 'foo_name';
+        $filter->type = 'stop';
+        $filter->options = [
+            'stopwords' => '_english_',
+        ];
+
+        $index->filters = [$filter];
+
+        $this->processor->process($this->documentMetadata, $index);
+
+        $this->assertEquals([
+            'settings' => [
+                'analysis' => [
+                    'filter' => [
+                        'foo_name' => [
+                            'type' => 'stop',
+                            'stopwords' => '_english_',
+                        ],
+                    ],
+                ],
+            ],
+        ], $this->documentMetadata->indexParams);
+    }
+
+    public function testTokenizersAreProcessedCorrectly()
+    {
+        $index = new Index();
+
+        $tokenizer = new Tokenizer();
+        $tokenizer->name = 'foo_name';
+        $tokenizer->type = 'ngram';
+        $tokenizer->options = [
+            'min_gram' => 3,
+        ];
+
+        $index->tokenizers = [$tokenizer];
+
+        $this->processor->process($this->documentMetadata, $index);
+
+        $this->assertEquals([
+            'settings' => [
+                'analysis' => [
+                    'tokenizer' => [
+                        'foo_name' => [
+                            'type' => 'ngram',
+                            'min_gram' => 3,
+                        ],
+                    ],
+                ],
+            ],
+        ], $this->documentMetadata->indexParams);
+    }
+
+    /**
+     * @group functional
+     */
+    public function testFunctional()
```

Comment:
void return type

this test name can be a little more verbose. I understand that is a functional but in this way I need to check in the method body what it test functionally.